### PR TITLE
Fixed issues in TM separated pattern -4

### PR DESCRIPTION
**Pull request by hasuniea**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: advanced/am-pattern-4/templates/am/traffic-manager/wso2am-pattern-4-am-trafficmanager-conf-entrypoint.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    echo "Start WSO2 Carbon server" >&2
+    if [[ -z "${PROFILE_NAME}" ]]
+    then
+      # start the server with the provided startup arguments
+      sh ${WSO2_SERVER_HOME}/bin/api-manager.sh "$@"
+    else
+      # start the server with the specified profile and provided startup arguments
+      sh ${WSO2_SERVER_HOME}/bin/api-manager.sh -Dprofile=${PROFILE_NAME} "$@"
+    fi
+
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/templates/am/traffic-manager/wso2am-pattern-4-am-trafficmanager-conf-entrypoint.yaml:62]:
```suggestion
```

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: advanced/am-pattern-4/requirements.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+dependencies:
+  - name: mysql-am
+    version: "4.0.0-1"
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/requirements.yaml:17]:
```suggestion
    version: "4.0.0-2"
```

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: advanced/am-pattern-4/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,82 @@
+{{/*
+Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/templates/_helpers.tpl:2]:
```suggestion
Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: advanced/am-pattern-4/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/Chart.yaml:1]:
```suggestion
# Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: advanced/am-pattern-4/requirements.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/requirements.yaml:1]:
```suggestion
# Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: advanced/am-pattern-4/templates/am/gateway/wso2am-pattern-4-am-gateway-conf.yaml [hunk 6]
```diff
@@ -0,0 +1,171 @@
+# Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/templates/am/gateway/wso2am-pattern-4-am-gateway-conf.yaml:1]:
```suggestion
# Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: advanced/am-pattern-4/templates/am/gateway/wso2am-pattern-4-am-gateway-deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/templates/am/gateway/wso2am-pattern-4-am-gateway-deployment.yaml:1]:
```suggestion
# Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: advanced/am-pattern-4/templates/am/gateway/wso2am-pattern-4-am-gateway-ingress.yaml [hunk 8]
```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/templates/am/gateway/wso2am-pattern-4-am-gateway-ingress.yaml:1]:
```suggestion
# Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: advanced/am-pattern-4/templates/am/gateway/wso2am-pattern-4-am-gateway-service.yaml [hunk 9]
```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/templates/am/gateway/wso2am-pattern-4-am-gateway-service.yaml:1]:
```suggestion
# Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: advanced/am-pattern-4/templates/am/traffic-manager/wso2am-pattern-4-am-trafficmanager-conf-entrypoint.yaml [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/templates/am/traffic-manager/wso2am-pattern-4-am-trafficmanager-conf-entrypoint.yaml:1]:
```suggestion
```

File: advanced/am-pattern-4/templates/wso2am-pattern-4-mi-secrets.yaml [hunk 11]
```diff
@@ -0,0 +1,30 @@
+{{ if and (not (eq .Values.wso2.subscription.username "")) (not (eq .Values.wso2.subscription.password "")) (eq (default "docker.wso2.com" .Values.wso2.deployment.mi.dockerRegistry) "docker.wso2.com") }}
+# Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/templates/wso2am-pattern-4-mi-secrets.yaml:2]:
```suggestion
# Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: advanced/am-pattern-4/templates/wso2am-pattern-4-secrets.yaml [hunk 12]
```diff
@@ -0,0 +1,30 @@
+{{ if and (not (eq .Values.wso2.subscription.username "")) (not (eq .Values.wso2.subscription.password "")) }}
+# Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/templates/wso2am-pattern-4-secrets.yaml:2]:
```suggestion
# Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: advanced/am-pattern-4/templates/wso2am-pattern-4-service-account.yaml [hunk 13]
```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/templates/wso2am-pattern-4-service-account.yaml:1]:
```suggestion
# Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "am-pattern-4.resource.prefix" . }}-mi-1-service
+  namespace : {{ .Release.Namespace }}
+  labels:
+    deployment: {{ template "am-pattern-4.resource.prefix" . }}-mi
+    node: {{ template "am-pattern-4.resource.prefix" . }}-mi-1
```
Comment (ThilinaManamgoda) [inline advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml:21]: The label value should be `{{ template "am-pattern-4.resource.prefix" . }}-mi`

File: advanced/am-pattern-4/values.yaml [hunk 15]
```diff
@@ -0,0 +1,322 @@
+# Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/values.yaml:1]:
```suggestion
# Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+      name: management
+    {{- if .Values.wso2.deployment.mi.synapseTest.enabled }}
+    - port: 9008
+      targetPort: 9008
+      protocol: TCP
+      name: synapse-test
+    {{- end}}
+  selector:
+    deployment: {{ template "am-pattern-4.resource.prefix" . }}-mi
+    node: {{ template "am-pattern-4.resource.prefix" . }}-mi-1
```
Comment (ThilinaManamgoda) [inline advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml:50]: The label value should be `{{ template "am-pattern-4.resource.prefix" . }}-mi`

File: README.md [hunk 17]
```diff
@@ -15,10 +15,11 @@
 * [Deployment Pattern 1](advanced/am-pattern-1/README.md)
 * [Deployment Pattern 2](advanced/am-pattern-2/README.md)
 * [Deployment Pattern 3](advanced/am-pattern-3/README.md)
+* [Deployment Pattern 4](advanced/am-pattern-4/README.md)
 
 ## Changelog
 
-**Change log** from previous `v3.2.0.4` release: [View Here](https://github.com/wso2/kubernetes-apim/blob/3.2.x/CHANGELOG.md)
+**Change log** from previous `v3.2.0.4` release: [View Here](https://github.com/wso2/kubernetes-apim/blob/3.2.x/CHANGELOG.md)git
```
Comment (chirangaalwis) [inline README.md:22]:
```suggestion
```

File: advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "am-pattern-4.resource.prefix" . }}-mi-1-service
```
Comment (ThilinaManamgoda) [inline advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml:17]: The name should be `{{ template "am-pattern-4.resource.prefix" . }}-mi-service`

File: README.md [hunk 19]
```diff
@@ -15,10 +15,11 @@
 * [Deployment Pattern 1](advanced/am-pattern-1/README.md)
 * [Deployment Pattern 2](advanced/am-pattern-2/README.md)
 * [Deployment Pattern 3](advanced/am-pattern-3/README.md)
+* [Deployment Pattern 4](advanced/am-pattern-4/README.md)
 
 ## Changelog
```
Comment (chirangaalwis) [inline README.md:20]:
```suggestion
```

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml (hunk 14 above)
Comment (hasuniea) [inline advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml:21]:
```suggestion
    node: {{ template "am-pattern-4.resource.prefix" . }}-mi
```

File: advanced/am-pattern-4/values.yaml [hunk 20]
```diff
@@ -0,0 +1,322 @@
...
+
+        # Kubernetes RollingUpdate strategy configurations
+        strategy:
+          rollingUpdate:
+            # The maximum number of pods that can be scheduled above the desired number of pods
+            maxSurge: 2
+            # The maximum number of pods that can be unavailable during the update
+            maxUnavailable: 0
+
+      # API Manager's Control Plane specific configurations
```
Comment (RakhithaRR) [inline advanced/am-pattern-4/values.yaml:144]: This should be TM, not CP

File: advanced/am-pattern-4/values.yaml [hunk 21]
```diff
@@ -0,0 +1,322 @@
...
+           jvm:
+             # Resource allocation for the Java Heap
+             heap:
+                memory:
+                # Initial and minimum Heap size
+                  xms: "1024m"
+                  # Maximum Heap size
+                  xmx: "1024m"
+
+        # If the deployment configurations for the Gateway profile of WSO2 API Manager v3.2.0 (<WSO2AM>/repository/conf/deployment.toml),
```
Comment (RakhithaRR) [inline advanced/am-pattern-4/values.yaml:181]: 4.0.0

File: advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml (hunk 18 above)
Comment (hasuniea) [inline advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml:17]:
```suggestion
  name: {{ template "am-pattern-4.resource.prefix" . }}-mi-service
```

File: advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml (hunk 14 above)
Comment (hasuniea) [inline advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml]:
```suggestion
    node: {{ template "am-pattern-4.resource.prefix" . }}-mi
```

File: advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml (hunk 16 above)
Comment (hasuniea) [inline advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml:50]:
```suggestion
    node: {{ template "am-pattern-4.resource.prefix" . }}-mi
```

Comment (hasuniea) [inline advanced/am-pattern-4/templates/mi/wso2am-pattern-4-mi-service.yaml:50]:
```suggestion
    node: {{ template "am-pattern-4.resource.prefix" . }}-mi
```

File: advanced/am-pattern-4/values.yaml (hunk 20 above)
Comment (hasuniea) [inline advanced/am-pattern-4/values.yaml:144]:
```suggestion
      # API Manager's Traffic Manager specific configurations
```

File: advanced/am-pattern-4/values.yaml (hunk 21 above)
Comment (hasuniea) [inline advanced/am-pattern-4/values.yaml:181]:
```suggestion
        # If the deployment configurations for the Gateway profile of WSO2 API Manager v4.0.0 (<WSO2AM>/repository/conf/deployment.toml),
```

File: advanced/am-pattern-4/values.yaml [hunk 22]
```diff
@@ -0,0 +1,322 @@
...
+        # add the customized configuration file under (wso2 -> deployment -> am -> gateway -> config -> deployment.toml)
+#        config:
+#          deployment.toml: |-
+#            # deployment configurations for the Gateway profile of WSO2 API Manager v3.2.0 (<WSO2AM>/repository/conf/deployment.toml)
```
Comment (hasuniea) [inline advanced/am-pattern-4/values.yaml:185]:
```suggestion
#            # deployment configurations for the Gateway profile of WSO2 API Manager v4.0.0 (<WSO2AM>/repository/conf/deployment.toml)
```

File: advanced/am-pattern-4/README.md [hunk 23]
```diff
@@ -0,0 +1,349 @@
...
+**Note:**
+
+* `NAMESPACE` should be the Kubernetes Namespace in which the resources are deployed.
+
+#### Install Chart From [WSO2 Helm Chart Repository](https://hub.helm.sh/charts/wso2)
+
+ Helm version 2
+
+ ```
+ helm install --name <RELEASE_NAME> wso2/am-pattern-4 --version 4.0.0-1 --namespace <NAMESPACE>
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/README.md:60]:
```suggestion
 helm install --name <RELEASE_NAME> wso2/am-pattern-4 --version 4.0.0-2 --namespace <NAMESPACE>
```

File: advanced/am-pattern-4/README.md [hunk 24]
```diff
@@ -0,0 +1,349 @@
...
+ ```
+
+ Helm version 3
+
+ - Deploy the Kubernetes resources using the Helm Chart
+ 
+    ```
+    helm install <RELEASE_NAME> wso2/am-pattern-4 --version 4.0.0-1 --namespace <NAMESPACE> --create-namespace
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/README.md:68]:
```suggestion
    helm install <RELEASE_NAME> wso2/am-pattern-4 --version 4.0.0-2 --namespace <NAMESPACE> --create-namespace
```

File: advanced/am-pattern-4/values.yaml [hunk 25]
```diff
@@ -0,0 +1,322 @@
...
+           jvm:
+             # Resource allocation for the Java Heap
+             heap:
+                memory:
+                # Initial and minimum Heap size
+                  xms: "1024m"
+                  # Maximum Heap size
+                  xmx: "1024m"
+
+        # If the deployment configurations for the Gateway profile of WSO2 API Manager v3.2.0 (<WSO2AM>/repository/conf/deployment.toml),
```
Comment (hasuniea) [inline advanced/am-pattern-4/values.yaml:181]:
```suggestion
        # If the deployment configurations for the Gateway profile of WSO2 API Manager v4.0.0 (<WSO2AM>/repository/conf/deployment.toml),
```

File: advanced/am-pattern-4/README.md [hunk 26]
```diff
@@ -0,0 +1,349 @@
...
+
+The above steps will deploy the deployment pattern using WSO2 product Docker images available at DockerHub.
+
+If you are using WSO2 product Docker images available from WSO2 Private Docker Registry,
+please provide your WSO2 Subscription credentials via input values (using `--set` argument). 
+
+Please see the following example.
+
+```
+ helm install --name <RELEASE_NAME> wso2/am-pattern-4 --version 4.0.0-1 --namespace <NAMESPACE> --set wso2.subscription.username=<SUBSCRIPTION_USERNAME> --set wso2.subscription.password=<SUBSCRIPTION_PASSWORD>
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/README.md:79]:
```suggestion
 helm install --name <RELEASE_NAME> wso2/am-pattern-4 --version 4.0.0-2 --namespace <NAMESPACE> --set wso2.subscription.username=<SUBSCRIPTION_USERNAME> --set wso2.subscription.password=<SUBSCRIPTION_PASSWORD>
```

File: advanced/am-pattern-4/README.md [hunk 27]
```diff
@@ -0,0 +1,349 @@
...
+```
+git clone https://github.com/wso2/kubernetes-apim.git
+```
+
+##### Deploy Helm chart for WSO2 API Manager Pattern 4 deployment.
+
+ Helm version 2
+
+ ```
+ helm install --dep-up --name <RELEASE_NAME> <HELM_HOME>/am-pattern-4 --version 4.0.0-1 --namespace <NAMESPACE>
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/README.md:100]:
```suggestion
 helm install --dep-up --name <RELEASE_NAME> <HELM_HOME>/am-pattern-4 --namespace <NAMESPACE>
```

File: advanced/am-pattern-4/README.md [hunk 28]
```diff
@@ -0,0 +1,349 @@
...
+ ```
+
+ Helm version 3
+
+ - Deploy the Kubernetes resources using the Helm Chart
+ 
+    ```
+    helm install <RELEASE_NAME> <HELM_HOME>/am-pattern-4 --version 4.0.0-1 --namespace <NAMESPACE> --dependency-update --create-namespace
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/README.md:108]:
```suggestion
    helm install <RELEASE_NAME> <HELM_HOME>/am-pattern-4 --namespace <NAMESPACE> --dependency-update --create-namespace
```

File: advanced/am-pattern-4/README.md [hunk 29]
```diff
@@ -0,0 +1,349 @@
...
+
+The above steps will deploy the deployment pattern using WSO2 product Docker images available at DockerHub.
+
+If you are using WSO2 product Docker images available from WSO2 Private Docker Registry,
+please provide your WSO2 Subscription credentials via input values (using `--set` argument). 
+
+Please see the following example.
+
+```
+ helm install --name <RELEASE_NAME> <HELM_HOME>/am-pattern-4 --version 4.0.0-1 --namespace <NAMESPACE> --set wso2.subscription.username=<SUBSCRIPTION_USERNAME> --set wso2.subscription.password=<SUBSCRIPTION_PASSWORD>
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/README.md:119]:
```suggestion
 helm install --name <RELEASE_NAME> <HELM_HOME>/am-pattern-4 --namespace <NAMESPACE> --set wso2.subscription.username=<SUBSCRIPTION_USERNAME> --set wso2.subscription.password=<SUBSCRIPTION_PASSWORD>
```

File: advanced/am-pattern-4/README.md [hunk 30]
```diff
@@ -0,0 +1,349 @@
...
+    ```
+    git clone https://github.com/wso2/kubernetes-apim.git
+    ```
+
+* Remove the MI templates by removing the `mi` folder in `<KUBERNETES_HOME>/advanced/am-pattern-4/templates/`.
+
+* Deploy Helm charts
+
+    ```helm
+    helm install <RELEASE_NAME> <HELM_HOME>/am-pattern-4 --version 4.0.0-1 --namespace <NAMESPACE> --dependency-update --create-namespace
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/README.md:348]:
```suggestion
    helm install <RELEASE_NAME> <HELM_HOME>/am-pattern-4 --version 4.0.0-2 --namespace <NAMESPACE> --dependency-update --create-namespace
```

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: advanced/am-pattern-4/README.md [hunk 31]
```diff
@@ -0,0 +1,349 @@
...
+### Choreo Analytics
+
+If you need to enable Choreo Analytics with WSO2 API Manager, please follow the documentation on [Register for Analytics](https://apim.docs.wso2.com/en/4.0.0/observe/api-manager-analytics/configure-analytics/register-for-analytics/) to obtain the on-prem key for Analytics.
+
+The following example shows how to enable Analytics with the helm charts.
+
+Helm v2
+
+```
+helm install --name <RELEASE_NAME> wso2/am-pattern-4 --version 4.0.0-1 --namespace <NAMESPACE> --set wso2.choreoAnalytics.enabled=true --set wso2.choreoAnalytics.endpoint=<CHOREO_ANALYTICS_ENDPOINT> --set wso2.choreoAnalytics.onpremKey=<ONPREM_KEY>
```
Comment (chirangaalwis) [inline advanced/am-pattern-4/README.md:147]:
```suggestion
helm install --name <RELEASE_NAME> wso2/am-pattern-4 --version 4.0.0-2 --namespace <NAMESPACE> --set wso2.choreoAnalytics.enabled=true --set wso2.choreoAnalytics.endpoint=<CHOREO_ANALYTICS_ENDPOINT> --set wso2.choreoAnalytics.onpremKey=<ONPREM_KEY>
```

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

**Review by chirangaalwis — APPROVED** [review]

File: advanced/am-pattern-4/README.md [hunk 32]
```diff
@@ -0,0 +1,349 @@
...
+    - `INTEGRATION_IMAGE_REGISTRY` will refer to the Docker registry that created Docker image has been pushed <br>
+    - `INTEGRATION_IMAGE_NAME` will refer to the name of the created Docker image <br>
+    - `INTEGRATION_IMAGE_TAG` will refer to the tag of the created Docker image <br><br>
+>2. If your Docker registry is a private registry, [create an imagePullSecret](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/).<br><br>
+    - `IMAGE_PULL_SECRET` will refer to the created image pull secret <br><br>
+>3. Deploy the helm resource using following command.<br><br>
+>   ```
+>   helm install <RELEASE_NAME> wso2/am-pattern-4 --version 4.0.0-1 --namespace <NAMESPACE> --set wso2.deployment.mi.dockerRegistry=<INTEGRATION_IMAGE_REGISTRY> --set wso2.deployment.mi.imageName=<INTEGRATION_IMAGE_NAME> --set wso2.deployment.mi.imageTag=<INTEGRATION_IMAGE_TAG> --set wso2.deployment.mi.imagePullSecrets=<IMAGE_PULL_SECRET>
```
Comment (hasuniea) [inline advanced/am-pattern-4/README.md:135]:
```suggestion
>   helm install <RELEASE_NAME> wso2/am-pattern-4 --version 4.0.0-2 --namespace <NAMESPACE> --set wso2.deployment.mi.dockerRegistry=<INTEGRATION_IMAGE_REGISTRY> --set wso2.deployment.mi.imageName=<INTEGRATION_IMAGE_NAME> --set wso2.deployment.mi.imageTag=<INTEGRATION_IMAGE_TAG> --set wso2.deployment.mi.imagePullSecrets=<IMAGE_PULL_SECRET>
```

**Review by RakhithaRR — APPROVED** [review]

**Review by ThilinaManamgoda — APPROVED** [review]

**Pull request merged**